### PR TITLE
Issue 39670: Have Location sort in alphabetical order for New Specimen Request page

### PR DESCRIPTION
**Pull request by labkey-adam**

### CODE REVIEW

File: study/src/org/labkey/study/model/StudyManager.java [hunk 1]
```diff
@@ -2950,8 +2928,6 @@ public void deleteAllStudyData(Container c, User user)
             assert deletedTables.add(StudySchema.getInstance().getTableInfoUploadLog());
             Table.delete(_datasetHelper.getTableInfo(), containerFilter);
             assert deletedTables.add(_datasetHelper.getTableInfo());
-//            Table.delete(_locationHelper.getTableInfo(), containerFilter);
```
Comment (labkey-jeckels) [inline study/src/org/labkey/study/model/StudyManager.java:2953]: Does this codepath clear the caches, including LocationCache? I can't find a place where it does, aside from calling clearCachedStudies() which is only a small part of what clearCaches() does

**Review by labkey-jeckels — APPROVED** [review]
One question on cache clearing, but feel free to merge without additional review

File: study/src/org/labkey/study/model/StudyManager.java (hunk 1 above)
Comment (labkey-adam) [reply study/src/org/labkey/study/model/StudyManager.java:2953]: You should verify this yourself, but I believe that each of those "helpers" is backed by a DbCache, which gets notified by Table whenever a mutating operation occurs on its TableInfo. Any insert, update, delete, or truncate results in clearing that entire cache. Which is why I hate DbCache... way too aggressive. I deprecated it a long time ago and have replaced some of the uses, but replacing these study helpers with a standard cache and explicit invalidation can be tricky.

The location tables are now provisioned (e.g., specimentables.c15d54_location) so I believe OnotologyManager takes care of deleting them. But you're right... OnotologManager doesn't know to clear the location cache. Adding it here might be too soon... but probably better than nothing. What do you think?

Comment (labkey-jeckels) [reply study/src/org/labkey/study/model/StudyManager.java:2953]: I verified that you're correct on the DbCache behavior.

OntologyManager is involved in deleting the domains via the call to SpecimenTablesProvider.deleteTables() so maybe we're OK as-is

Comment (labkey-adam) [reply study/src/org/labkey/study/model/StudyManager.java:2953]: Good catch... I think it probably makes sense to clear the location cache immediately after the call to deleteTables() then. The only bad scenario I can think of is: delete study followed by create or import study without any locations... without the clear, the old ones would be left hanging around.

Comment (labkey-adam) [reply study/src/org/labkey/study/model/StudyManager.java:2953]: Add a clear after deleteTables()

**Pull request merged**